### PR TITLE
Add eslint to node bindings

### PR DESCRIPTION
**Pull request by matthewkeil**
No code changes were included.

_**note:** I coped the eslint configuration from `@chainsafe/blst-ts`_

## Inclusions
- Add industry standard `eslint`
- Run format
- Add command to makefile

### CODE REVIEW

**Comment by dapplion** [comment]
If you removed Typescript to keep the bindings less verbose in terms of tooling, adding eslint kind of completely reverses that idea right? 
- referencing https://github.com/ethereum/c-kzg-4844/pull/260

**Comment by matthewkeil** [comment]
I was talking with @jtraglia and he asked about some linting in the `C` code so I put up #268 and asked if I should add eslint for the TS/JS.  I do not feel strongly about this inclusion and am ok with either merging or closing this.  What are your thoughts @asn-d6

File: bindings/node.js/Makefile [hunk 1]
```diff
@@ -61,7 +61,7 @@ test: install
 # Lint js/ts code
 .PHONY: format
 format: install
-	@$(YARN) prettier --loglevel=warn --write .
+	@yarn eslint --quiet --color --ext .ts lib/ test/
```
Comment (jtraglia) [inline bindings/node.js/Makefile:64]: I'd like for this to use the `$(YARN)` command to hide extraneous outputs.
```suggestion
	@$(YARN) eslint --quiet --color --ext .ts lib/ test/
```

Without it, I get this, which I don't find all that useful:
```
yarn run v1.22.19
$ /Users/jtraglia/Projects/mattk/c-kzg-4844/bindings/node.js/node_modules/.bin/eslint --quiet --color --ext .ts lib/ test/
```

**Review by jtraglia — CHANGES_REQUESTED** [review]
Nice, I like this. Sure, setting up eslint is complicated, but the code looks better IMO.

Just a couple of small nits in the comments.

File: bindings/node.js/package.json [hunk 2]
```diff
@@ -13,6 +13,12 @@
   },
   "devDependencies": {
     "@types/jest": "^29.2.1",
+    "@typescript-eslint/eslint-plugin": "^4.21.0",
+    "@typescript-eslint/parser": "^4.21.0",
+    "eslint": "^7.23.0",
+    "eslint-plugin-import": "^2.22.1",
+    "eslint-plugin-node": "^11.1.0",
+    "eslint-plugin-prettier": "^3.3.1",
```
[the remark this answers is not in the document]
Comment (jtraglia) [reply bindings/node.js/package.json]: Thanks, that warning is gone now 👍

**Comment by matthewkeil** [comment]
There are merge conflicts from the last merge and there will be another round when the #268 gets merged.  I will fix them all in one go after that one is good to go.

**Review by dapplion — APPROVED** [review]

**Comment by asn-d6** [comment]
Let's do it!

**Pull request merged**